compiler/visitor.h

```diff
@@ -49,17 +49,47 @@ class Visitor : public ifccBaseVisitor
 
   virtual antlrcpp::Any visitDeclaration(ifccParser::DeclarationContext *context) override
   {
-    int variablesNumber = context->VARIABLENF().size() + 1; // +1 for the final variable
-    int variableOffset = variablesNumber * 4;               // initializes the highest offset for the first variable
-    this->maxOffset = variableOffset;
+    int variablesNumber = context-> declarationvar().size();
+    int variableOffset = variablesNumber*4; // initializes the highest offset for the first variable
     std::map<std::string, int> symbolTable; // SymbolTable
-    for (int i = 0; i < variablesNumber - 1; i++)
-    {
+    this->symbolTable = symbolTable; // Copy the symbolTable for the whole visitor object
+    this->variableOffset = variableOffset;
+    this->maxOffset = variableOffset;
+    visitChildren(context);
+    /*for(int i=0; i<variablesNumber-1; i++) {
       symbolTable.insert({removeLastCharFromString(context->VARIABLENF()[i]->getText()), variableOffset});
       variableOffset -= 4;
     }
     symbolTable.insert({context->VARIABLE()->getText(), variableOffset});
     this->symbolTable = symbolTable; // Copy the symbolTable for the whole visitor object
+    */
+    return 0;
+  }
+
+  virtual antlrcpp::Any visitDeclarationSeule(ifccParser::DeclarationSeuleContext *context) override
+  {
+    symbolTable.insert({context->VARIABLE()->getText(), variableOffset});
+    this->variableOffset -=4;
+    return 0;
+  }
+
+  virtual antlrcpp::Any visitDeclarationInitialiseeConst(ifccParser::DeclarationInitialiseeConstContext *context) override
+  {
+    symbolTable.insert({context->VARIABLE()->getText(), variableOffset});
+    this->variableOffset -=4;
+    int varValue = stoi(context->CONST()->getText());
+    std::cout << "\tmovl $" << varValue << ", -" << this->symbolTable[context->VARIABLE()->getText()] << "(%rbp)" << std::endl;
```

Comment:
Add a test on this->symbolTable[context->VARIABLE()->getText()] (what if does not exist in the symbol table?)